src/rdb.c

```diff
@@ -3562,6 +3564,10 @@ void backgroundSaveDoneHandler(int exitcode, int bysignal) {
     /* Possibly there are replicas waiting for a BGSAVE in order to be served
      * (the first stage of SYNC is a bulk transfer of dump.rdb) */
     updateReplicasWaitingBgsave((!bysignal && exitcode == 0) ? C_OK : C_ERR, type);
+
+    /* Slot export should also be notified, in case this was a export related
+     * snapshot */
+    clusterHandleSlotExportBackgroundSaveDone((!bysignal && exitcode == 0) ? C_OK : C_ERR);
```

Comment:
similar to my comment in networking.c. I think rdb.c should be a dependent of cluster_migrateslot.c as opposed to the other way around. let's take a close and holistic look at the layering in rc2.

Reply:
Yeah - if you look above - this is basically just following the pattern we used for replication. I think we should invest in decoupling both this replication from RDB. But it is an orthogonal task to ASM